### PR TITLE
Search functionality

### PR DESCRIPTION
**Pull request by danachen**
## Description

Search for items on the shopping list by typing any part of the item name. Pressing on the UI "x" should re-render the list.

## Acceptance Criteria

- Display a text field above the top of the shopping list
- As the user types into the field, the list should narrow to display only items that contain the text the user entered in the filter field
- When the field has text in it, the user should be able to tap a UI element (e.g., with an "X" button next to the field) to clear the field
- The filter text should match any part of the item name (i.e. it should not only match from the start of the string)

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Testing Steps / QA Criteria

- Create a new list, and add items to the list
- Type in item on the list, the list should be filtered based on input
- Pressing the "x" inside the search bar should white-out the search field
- The list should re-render in its entirety (this is the part not yet working)

### CODE REVIEW

**Comment by adiati98** [comment]
@danachen & @fatima-ola,
I've tested the PR, and it works perfectly on my end!
The clear button does clear the search bar and also render the whole list. 
Great job! 🎉

great job with Eddie Murphy giphy

File: src/components/ListItem.js [hunk 1]
```diff
@@ -43,10 +40,11 @@ const ListItem = () => {
         onSnapshot(doc(itemsCollectionRef, currToken), (doc) => {
           if (!doc.data()) {
             setEmptyList(true);
+            setLoading(false);
           } else {
             setItems(doc.data().items);
+            setLoading(false);
```
Comment (llunaplanet) [inline src/components/ListItem.js]: This is an interesting change, the old code should have the same functionality, I'm curious, what problem were you solving with this change?

File: src/components/ListItem.js [hunk 2]
```diff
@@ -76,10 +76,10 @@ const ListItem = () => {
   const searchItems = async (e) => {
     let docSnap = await getDoc(currentCollectionRef);
     const data = docSnap.data().items;
-    const filteredItems = [];
+    let filteredItems = [];
 
     for (const item of data) {
-      console.log(item);
+      if (!e.target.value.length) filteredItems = data;
```
Comment (llunaplanet) [inline src/components/ListItem.js]: There is a refactor technique to deal with nested **if** clauses and improve the readability of the code, it's called **guard clauses**, you can read about it here https://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html

**Review by llunaplanet — APPROVED** [review]
Great work, all the AC are met so LGTM!

Great to see the evolution of the code in the commit history, going from low level **regexp** to using language provided functions like **.includes**. Also you started to extract functionality that can be reused to stand alone files 👍 

I have added a couple of non blocking comments, you can merge when you are ready.

File: src/components/ListItem.js (hunk 2 above)
Comment (llunaplanet) [reply src/components/ListItem.js]: ( I know that that code is not part of the final PR, so it's only **FYI**

**Pull request merged**